packages/saber/lib/cli.js

```diff
@@ -11,10 +12,15 @@ cli
   .option('--lazy', 'Enable lazy page compilation')
   .option('--port <port>', 'Server port', { default: 3000 })
   .option('--host <host>', 'Server host', { default: '0.0.0.0' })
-  .action((cwd = '.', options) => {
+  .action(async (cwd = '.', options) => {
     setNodeEnv('development')
 
-    const { host, port, lazy } = options
+    let { host, port, lazy } = options
+    host = host ? host : '0.0.0.0'
+    if (!port) {
+      port = await getPort({ port: getPort.makeRange(3000, 4000), host })
+    }
```

Comment:
let's validate the `port` in `index.js` instead, because user can also set `port` from `saber-config.js`